File: JAVA_COURSE/JAVA/16_4_Robot4_queue/src/edu/javacourse/robot/RobotManager.java
package edu.javacourse.robot;

import edu.javacourse.robot.ui.RobotFrame;
import javax.swing.JFrame;

public class RobotManager
{
    public static void main(String[] args) {
        // Количество сторон многоугольника
        final int COUNT = 7;
        // Длина стороны
        final int SIDE = 200;
        
        Robot robot;
        robot = new Robot(200, 50);
        // Создаем замкнутую фигуру с количеством углов COUNT
        for (int i = 0; i < COUNT; i++) {
            robot.forward(SIDE);
            robot.setCourse(robot.getCourse() + 360 / COUNT);
        }

        // Создаем форму для отрисовки пути нашего робота
        RobotFrame rf;
        rf= new RobotFrame(robot);
        rf.setDefaultCloseOperation(JFrame.EXIT_ON_CLOSE);
        rf.setVisible(true);
    }
}

File: JAVA_COURSE/KOTLIN/20_1_Robot5/src/robotkotlin/RobotManager.kt
package robotkotlin

import robotkotlin.ui.RobotFrame

import javax.swing.*

object RobotManager {

    @JvmStatic
    fun main(args: Array<String>) {
        // Количество сторон многоугольника
        val COUNT = 14
        // Длина стороны
        val SIDE = 100

        val robot = Robot(200.0, 50.0)
        // Установка слушателя для робота
        val srl = SimpleRobotListener()
        robot.setListener(srl)
        // Создаем замкнутую фигуру с количеством углов COUNT
        for (i in 0 until COUNT) {
            robot.forward(SIDE)
            robot.course = robot.course + 360 / COUNT
        }

        // Создаем форму для отрисовки пути нашего робота
        val rf = RobotFrame(robot)
        rf.defaultCloseOperation = JFrame.EXIT_ON_CLOSE
        rf.isVisible = true
    }
}
